osu.Game/Screens/Ranking/ResultsScreen.cs

```diff
@@ -204,17 +205,31 @@ protected override void Update()
 
             if (lastFetchCompleted)
             {
-                APIRequest nextPageRequest = null;
+                if (ScorePanelList.IsEmpty)
+                {
+                    // This can happen if a beatmap part of a playlist hasn't been played yet.
+                    VerticalScrollContent.Add(new OsuSpriteText
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Font = OsuFont.GetFont(size: 32, weight: FontWeight.Regular),
+                        Text = "no scores yet!",
+                    });
+                }
+                else
```

Comment:
If the score panel list is initially empty, there shouldn't be any more requests. `IsScrolledToStart` and `IsScrolledToEnd` check if `flow.Count > 0`.